src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util.platforms.windows;
+
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.util.Utils;
+import com.aws.greengrass.util.platforms.Exec;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+@SuppressWarnings("PMD.AvoidCatchingThrowable")
+public class WindowsExec extends Exec {
+    public static final String PATHEXT_KEY = "PATHEXT";
+
+    private static final Logger logger = LogManager.getLogger(WindowsExec.class);
+    private static final List<String> PATHEXT;  // ordered file extensions to try, when no extension is provided
```

Comment:
why? Is this some standard?

`PATH` is a valid environment variable on windows AFAIK

Reply:
PATHEXT is for completing the executable file extension when none is provided. See remarks [here](https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2012-r2-and-2012/cc753148(v=ws.11))

Reply:
oh, I see